src/main/java/randoop/generation/constanttfidf/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,298 @@
+    Object scope;
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        throw new RuntimeException("Should not update totalClasses in CLASS level");
+      case PACKAGE:
+        scope = ((ClassOrInterfaceType) type).getPackage();
+        break;
+      case ALL:
+        scope = null;
+        break;
+      default:
+        throw new RuntimeException("Unknown literals level");
+    }
+    scopeStatisticsMap
+        .computeIfAbsent(scope, __ -> new ScopeStatistics())
+        .addToTotalClasses(numClasses);
+  }
+
+  /**
+   * Get all sequences that had been recorded under the specific scope, which are the constants
+   * extracted by constant mining.
+   *
+   * @param scope the specific package, class, or null
+   * @return the set of sequences that have been recorded under the specific scope
+   */
+  public Set<Sequence> getSequencesForScope(Object scope) {
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("The scope %s is not found in the frequency information", scope);
+      return new HashSet<>();
+    }
+
+    return stats.getSequenceSet();
+  }
+
+  /**
+   * Get the frequency information for all scopes.
+   *
+   * @return the frequency information for all scopes
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumUses() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumUses()));
+    return res;
+  }
+
+  /**
+   * Get the frequency information of the given scope.
+   *
+   * @param scope a type, a package, or null
+   * @return the frequency information of the given scope
+   */
+  public Map<Sequence, Integer> getNumUses(Object scope) {
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        return scopeStatisticsMap.get((ClassOrInterfaceType) scope).getNumUses();
+      case PACKAGE:
+        return scopeStatisticsMap.get((Package) scope).getNumUses();
+      case ALL:
+        return getNumUses().get(null);
+      default:
+        throw new RandoopBug("Unexpected literals level: " + GenInputsAbstract.literals_level);
+    }
+  }
+
+  /**
+   * Get the numClassesWith information for all scopes.
+   *
+   * @return the numClassesWith information for all scopes
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumClassesWith() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumClassesWith()));
+    return res;
+  }
+
+  /**
+   * Get the numClassesWith information of the specific scope.
+   *
+   * @param scope the specific scope
+   * @return the numClassesWith information of the specific scope
+   */
+  public Map<Sequence, Integer> getNumClassesWith(Object scope) {
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        throw new RandoopBug("Should not get classesWithConstant in CLASS level");
```

Comment:
Similarly to the above, I think this method can be a check against CLASS, and then (if no exception is thrown) a one-liner.